### PR TITLE
Amend test data generation

### PR DESCRIPTION
**Pull request by richardpattinson**
## Context

We are adding permissions randomisation on test data generation, additionally we are ensuring that the other providers have permissions for the test provider.
 
## Link to Trello card

https://trello.com/c/aOn0TOSq/3767-amend-test-data-generation

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/lib/create_example_provider_users_with_permissions_spec.rb [hunk 1]
```diff
@@ -8,6 +8,20 @@
...
+      create(:provider, code: code)
+    end
+    CreateExampleProviderUsersWithPermissions.call
+    permissions = ProviderRelationshipPermissions.all
+    expect(permissions.first.ratifying_provider_can_make_decisions).to be(true)
+    expect(permissions.first.ratifying_provider_can_view_safeguarding_information).to be(false)
+    expect(permissions.first.training_provider_can_view_diversity_information).to be(true)
+    expect(permissions.last.training_provider_can_make_decisions).to be(false)
+    expect(permissions.last.training_provider_can_view_safeguarding_information).to be(false)
+    expect(permissions.last.ratifying_provider_can_view_diversity_information).to be(false)
```
Comment (richardpattinson) [inline spec/lib/create_example_provider_users_with_permissions_spec.rb:25]: We are not sure if this is the best way to test randomness?

File: app/lib/create_example_provider_users_with_permissions.rb [hunk 2]
```diff
@@ -44,9 +44,12 @@ def self.call
...
-      training_provider_can_make_decisions: true,
-      training_provider_can_view_safeguarding_information: true,
-      training_provider_can_view_diversity_information: true,
+      ratifying_provider_can_make_decisions: [true, false].sample,
+      ratifying_provider_can_view_safeguarding_information: [true, false].sample,
+      ratifying_provider_can_view_diversity_information: [true, false].sample,
+      training_provider_can_make_decisions: [true, false].sample,
+      training_provider_can_view_safeguarding_information: [true, false].sample,
+      training_provider_can_view_diversity_information: [true, false].sample,
       setup_at: Time.zone.now,
```
Comment (despo) [inline app/lib/create_example_provider_users_with_permissions.rb:53]: Can we use a FactoryBot to achieve this instead, similar to https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/lib/test_applications.rb? There you can use Faker to retrieve a random value for each, using `Faker::Boolean.boolean(true_ratio: 0.5)`

**Comment by despo** [comment]
Can we discuss this after the standup tomorrow? The issue is that the `ProviderRelationshipPermissions` relationship between course providers and accredited providers is missing, as users will be prompted to setup permissions if the relationship is there but not setup, so I don't think this will resolve the issue that we have encountered.

**Comment by cwrw** [comment]
So I don't think we need the random changes where we are already creating the `ProviderRelationshipPermissions`. However a new creation for `ProviderRelationshipPermissions` has been added for generating the fake test provider, but yea let's catchup with both

File: app/lib/create_example_provider_users_with_permissions.rb (hunk 2 above)
Comment (richardpattinson) [reply app/lib/create_example_provider_users_with_permissions.rb:53]: Sure thing, happy to chat

**Comment by richardpattinson** [comment]
> Can we discuss this after the standup tomorrow? The issue is that the `ProviderRelationshipPermissions` relationship between course providers and accredited providers is missing, as users will be prompted to setup permissions if the relationship is there but not setup, so I don't think this will resolve the issue that we have encountered.

Sure thing, lets chat after the standup

**Review by cwrw — APPROVED** [review]
Thanks for addressing comments!

**Pull request merged**